app/notification.go

```diff
@@ -578,47 +610,67 @@ func getMentionsEnabledFields(post *model.Post) model.StringArray {
 	return ret
 }
 
+// allowChannelMentions returns whether or not the channel mentions are allowed for the given post.
+func (a *App) allowChannelMentions(post *model.Post, numProfiles int) bool {
+	if post.Type == model.POST_HEADER_CHANGE || post.Type == model.POST_PURPOSE_CHANGE {
+		return false
+	}
+
+	if int64(numProfiles) >= *a.Config().TeamSettings.MaxNotificationsPerChannel {
```

Comment:
I've seen this condition several times through this change, we should probably consider naming it

Reply:
I'll reduce the duplication of code when it's used in `SendNotifications`, but that's not going to change how it's used here.